deadlock_demolition.md

```diff
@@ -0,0 +1,55 @@
+---
+layout: doc
+title: "Deadlock Demolition"
+learning_objectives:
+  - Synchronization Primitives
+  - Deadlock Detection using a Resource Allocation Graph
+---
+
+## Overview
+
+This week, you will be building a library for deadlock-resistant mutex (drm) locks. A drm should behave like a pthread_mutex_t but not allow itself to be locked by a thread if the attempt to lock it would result in deadlock. We will use binary semaphore notation for your lock's functions:i.e. post is equivalent to unlocking and wait is equivalent to locking.
```

Comment:
Also mention `PTHREAD_MUTEX_ERRORCHECK` and `PTHREAD_MUTEX_RECURSIVE` as related concepts that are fun to know. Maybe in a separate section at the bottom as a side note?

Reply:
I don't know much about those things.
Do you want anything more than just 'check these out' and links?